File: src/main/java/com/atsuishio/superbwarfare/init/ModAttributes.java
package com.atsuishio.superbwarfare.init;

import com.atsuishio.superbwarfare.Mod;
import net.minecraft.world.entity.ai.attributes.Attribute;
import net.minecraft.world.entity.ai.attributes.RangedAttribute;

import java.util.function.Supplier;

public class ModAttributes {
    public static final Supplier<Attribute> BULLET_RESISTANCE = Registration.attribute("bullet_resistance",
            () -> new RangedAttribute("attribute." + Mod.MODID + ".bullet_resistance", 0, 0, 1).setSyncable(true));
    public static final Supplier<Attribute> BLOCK_REACH = Registration.attribute("block_reach",
            () -> new RangedAttribute("attribute." + Mod.MODID + ".block_reach", 4.5, 0, 1024).setSyncable(true));
    public static final Supplier<Attribute> ENTITY_REACH = Registration.attribute("entity_reach",
            () -> new RangedAttribute("attribute." + Mod.MODID + ".entity_reach", 3, 0, 1024).setSyncable(true));

    public static void register() {
        BULLET_RESISTANCE.get();
        BLOCK_REACH.get();
        ENTITY_REACH.get();
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/init/ModAttributes.kt
package com.atsuishio.superbwarfare.init

import com.atsuishio.superbwarfare.Mod
import io.github.fabricators_of_create.porting_lib.attributes.PortingLibAttributes
import net.minecraft.world.entity.ai.attributes.Attribute
import net.minecraft.world.entity.ai.attributes.RangedAttribute

object ModAttributes {
    @JvmField
    val BULLET_RESISTANCE: Attribute = Registration.attribute("bullet_resistance") {
        RangedAttribute("attribute.${Mod.MODID}.bullet_resistance", 0.0, 0.0, 1.0).setSyncable(true)
    }

    @JvmField
    val BLOCK_REACH: Attribute = PortingLibAttributes.BLOCK_REACH

    @JvmField
    val ENTITY_REACH: Attribute = PortingLibAttributes.ENTITY_REACH

    @JvmStatic
    fun init() = Unit
}
